### PR TITLE
Pass SSL options to python triton client instantiations

### PR DESCRIPTION
**Pull request by matthewkotila**

### CODE REVIEW

**Comment by matthewkotila** [comment]
I ran both gRPC and HTTP locally, and MA correctly passes SSL options to python client

File: model_analyzer/triton/client/http_client.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (tanmayv25) [inline model_analyzer/triton/client/http_client.py:1]: You can remove `(c)` to make the line shorter.

File: model_analyzer/triton/client/client_factory.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (tanmayv25) [inline model_analyzer/triton/client/client_factory.py:1]: remove `(c)`

File: model_analyzer/triton/client/client_factory.py [hunk 3]
```diff
@@ -23,29 +23,33 @@ class TritonClientFactory:
...
     @staticmethod
-    def create_grpc_client(server_url):
+    def create_grpc_client(server_url, ssl_options={}):
         """
         Parameters
         ----------
         server_url : str
             The url for Triton server's GRPC endpoint
+        ssl_options : dict
+            Dictionary of SSL options for gRPC python client
```
Comment (tanmayv25) [inline model_analyzer/triton/client/client_factory.py:33]: Don't we send the entire dict here? Inclusive of https? As there is a single `get_ssl_options` function. Would it make more sense to have `get_https_ssl_options` and `get_grpc_ssl_options`?

File: model_analyzer/triton/client/http_client.py [hunk 4]
```diff
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import gevent.ssl
```
Comment (tgerdesnv) [inline model_analyzer/triton/client/http_client.py:15]: Do we need to do anything to make sure this is available in the container and/or wheel?

File: model_analyzer/triton/client/client_factory.py [hunk 5]
```diff
@@ -23,29 +23,33 @@ class TritonClientFactory:
...
     @staticmethod
-    def create_http_client(server_url):
+    def create_http_client(server_url, ssl_options={}):
         """
         Parameters
         ----------
         server_url : str
             The url for Triton server's HTTP endpoint
+        ssl_options : dict
+            Dictionary of SSL options for HTTP python client
```
Comment (tanmayv25) [inline model_analyzer/triton/client/client_factory.py:49]: Same as above.

File: model_analyzer/triton/client/grpc_client.py [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (tanmayv25) [inline model_analyzer/triton/client/grpc_client.py:1]: remove `(c)`

File: model_analyzer/triton/client/http_client.py [hunk 7]
```diff
@@ -22,12 +25,62 @@ class TritonHTTPClient(TritonClient):
...
+                    'ssl-https-private-key-file']
+            else:
+                logging.warning(
+                    'model-analyzer with SSL must be passed a private key file in PEM format.'
+                )
+        if 'ssl-https-verify-peer' in ssl_options:
+            verify_peer = ssl_options['ssl-https-verify-peer']
+        if 'ssl-https-verify-host' in ssl_options:
+            verify_host = ssl_options['ssl-https-verify-host']
+        if verify_peer == 0 and verify_host == 0:
```
Comment (tanmayv25) [inline model_analyzer/triton/client/http_client.py:76]: Shouldn't this be `or` condition?

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

File: model_analyzer/triton/client/http_client.py (hunk 4 above)
Comment (matthewkotila) [reply model_analyzer/triton/client/http_client.py:15]: @tanmayv25 is there a particular version or set of versions of [gevent](https://pypi.org/project/gevent/) that we need?

File: model_analyzer/triton/client/client_factory.py (hunk 3 above)
Comment (debermudez) [reply model_analyzer/triton/client/client_factory.py:33]: At the moment, I think we leave as is.
Can you create a ticket to fix this later?

File: model_analyzer/triton/client/http_client.py (hunk 1 above)
Comment (matthewkotila) [reply model_analyzer/triton/client/http_client.py:1]: I think it's fine to leave the `(c)` in to follow guidance from company, as long as this fits on one line with the auto-formatter, which it does.

File: model_analyzer/triton/client/client_factory.py (hunk 2 above)
Comment (matthewkotila) [reply model_analyzer/triton/client/client_factory.py:1]: I think it's fine to leave the `(c)` in to follow guidance from company, as long as this fits on one line with the auto-formatter, which it does.

File: model_analyzer/triton/client/grpc_client.py (hunk 6 above)
Comment (matthewkotila) [reply model_analyzer/triton/client/grpc_client.py:1]: I think it's fine to leave the (c) in to follow guidance from company, as long as this fits on one line with the auto-formatter, which it does.

File: model_analyzer/triton/client/client_factory.py (hunk 3 above)
Comment (matthewkotila) [reply model_analyzer/triton/client/client_factory.py:33]: Refactored to two functions 👍

File: model_analyzer/triton/client/client_factory.py (hunk 5 above)
Comment (matthewkotila) [reply model_analyzer/triton/client/client_factory.py:49]: Refactored to two functions 👍

File: model_analyzer/triton/client/http_client.py (hunk 7 above)
Comment (matthewkotila) [reply model_analyzer/triton/client/http_client.py]: Nice catch--changed to `or` 👍

File: model_analyzer/triton/client/http_client.py (hunk 4 above)
Comment (matthewkotila) [reply model_analyzer/triton/client/http_client.py:15]: i set it to require `gevent>=21.12.0` 👍

**Review by debermudez — APPROVED** [review]

File: model_analyzer/triton/client/grpc_client.py (hunk 6 above)
Comment (dzier) [reply model_analyzer/triton/client/grpc_client.py:1]: The correct string is `Copyright 20xx-20yy, NVIDIA CORPORATION & AFFILIATES. All rights reserved.` So Tanmay is correct here.

File: model_analyzer/triton/client/http_client.py (hunk 1 above)
Comment (tgerdesnv) [reply model_analyzer/triton/client/http_client.py:1]: agreed

Comment (dzier) [reply model_analyzer/triton/client/http_client.py:1]: The correct string is Copyright 20xx-20yy, NVIDIA CORPORATION & AFFILIATES. All rights reserved. So Tanmay is correct here.

Comment (matthewkotila) [reply model_analyzer/triton/client/http_client.py]: updated to remove `(c)` 👍

**Review by tgerdesnv — APPROVED** [review]

**Pull request closed without merging**